tests/templates/test_subroutines/test_select.py

```diff
@@ -836,3 +839,85 @@ def circuit():
         probs = circuit()
         exp = np.eye(4)[0]
         assert np.allclose(probs, exp)
+
+
+@pytest.mark.parametrize("partial", [False, True])
+class TestSelectWithWorkWire:
+    """Tests for the 1 work-wire decomposition of Select."""
+
+    @pytest.mark.parametrize("num_controls", [0, 1, 2, 3])
+    def test_no_ops(self, num_controls, partial):
+        """Test that the decomposition does not return any operators for an empty list
+        of target operators."""
+
+        control = list(range(num_controls))
+        work = ["a"]
+
+        with qml.tape.QuantumTape() as tape:
+            _select_decomp_multi_control_work_wire(
+                ops=[], control=control, work_wires=work, partial=partial
+            )
+
+        assert tape.operations == []
```

Comment:
core told me that we should use queues instead of tapes.
```suggestion
        with qml.queuing.AnnotatedQueue() as q:
            _select_decomp_multi_control_work_wire(
                ops=[], control=control, work_wires=work, partial=partial
            )

        assert len(q) == 0
```